### DISCUSSION TITLE
I need help for component - DateRangePicker

### DISCUSSION THREAD

**Discussion by Nizhurin** [rsuite/rsuite#2597] [General]
Hello, Dear Team RSuite. First of all I thank You for the components library it's very nice. I need advice or help 
![VnzX1](https://user-images.githubusercontent.com/12280870/179456802-8637fd3b-dd52-4b42-87a1-7ed0a6a4bcc4.png)

Here I picked the date range of Apr 22 to May 4. But notice the time of day. In order to set those to 00:00:00 and 23:59:59 respectively, I have to click on those times and scroll around to pick specific time of day, which is super annoying to do every time.
[The documentation](https://rsuitejs.com/components/date-range-picker/) says nothing about the time. Also their [GitHub issues](https://github.com/rsuite/rsuite/issues) page does not address this. What am I missing here. It seems like such an obvious oversight to not allow you to set a default time of day start to 00:00:00 and end to 23:59:59.

Thank you for help or advice.

**Comment by simonguo** [comment]
This is indeed a design not considered. I was going to use `defaultCalendarValue` to set a default time range to suit this scenario, but found that there is still a problem, waiting for us to fix it.

```tsx
<DateRangePicker
  format="yyyy-MM-dd HH:mm:ss"
  defaultCalendarValue={[new Data('2022-07-01 00:00:00'), new Data('2022-08-01 23:59:59')]}
/>
```

**Reply by Nizhurin** [reply]
Thank You, I waiting for fix

**Reply by simonguo** [reply]
Fixed in [5.21.0](https://github.com/rsuite/rsuite/releases/tag/v5.21.0)

**Comment by alive2020** [comment]
Hi~
I've encountered the same issue:

Here I did preset for the value={startDay with 00:00:00 time, and endDay with 23:59:59 time}
But as soon as I open DateRangePicker and select startDay 19th of Oct it auto reset the endDay time to 00:00:00

![Screen Shot 2022-10-26 at 5 27 25 PM](https://user-images.githubusercontent.com/68528319/197976049-54c0170d-3cde-42cb-b651-60044025f49e.png)

![Screen Shot 2022-10-26 at 5 30 11 PM](https://user-images.githubusercontent.com/68528319/197976100-7c6710f9-e26a-4379-8da2-48d0c7d1b83b.png)

In the video below you can check it ~
It autosets the endDay time with the time of startDay.

https://user-images.githubusercontent.com/68528319/197977274-0233175c-5b60-4deb-b2f5-a2f2ebfada4b.mov


It would be just great to have a fix of this issue. @simonguo 
Thank you.

**Reply by simonguo** [reply]
1. What is the version of rsuite used?
2. Is the defaultCalendarValue property used?

Hope the following example can provide you with reference.
https://codesandbox.io/s/stoic-sun-fgw8er

**Reply by alive2020** [reply]
We are using
1. "rsuite": "^5.19.1",
2. Yes, I also tried to use defaultCalendarValue, 
but With or Without the problem is there:

As soon as I choose the first Date the default time of the second Date(endDay) 23:59:59 resets to 00:00:00.

**Reply by simonguo** [reply]
https://codesandbox.io/s/stoic-sun-fgw8er 
Can it be reproduced in the example?

**Reply by alive2020** [reply]
Yes, I'm attaching screenshot video :

https://user-images.githubusercontent.com/68528319/198009008-1940cf40-4b93-4ed3-9114-2cbaa4db60d4.mov

So when I select 7th of Feb and then select the 10th , the preset time  from 2pm changes to 12pm.
Is should stay as it was 2pm until I click and change it.

**Reply by simonguo** [reply]
See, we'll fix it asap.

**Reply by alive2020** [reply]
Thank you ^^
I also reported a bug related to DateRangePicker - https://github.com/rsuite/rsuite/issues/2839
Hope you will have time to review it.

Thank you. @simonguo

**Reply by simonguo** [reply]
Fixed in https://github.com/rsuite/rsuite/pull/2841

**Reply by simonguo** [reply]
https://codesandbox.io/s/compassionate-swanson-miozhb
The above is an example after the fix, to see if it solves your problem.

**Reply by alive2020** [reply]
I installed rsuite ver `5.21.0`, but my problem wasn't fixed. I encounter the same problem as it was before, as soon as I pick the second date the time automatically changes (it sets the same time as it is on StartDay)
and on this ex https://codesandbox.io/s/compassionate-swanson-miozhb - too, nothing has changed.

**Reply by simonguo** [reply]
You try to upgrade to a version later than [5.22.1](https://github.com/rsuite/rsuite/releases/tag/v5.22.1).
